dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml

```diff
@@ -178,14 +178,13 @@
         select t.state, count(0) as count
         from t_ds_process_instance t
         join t_ds_process_definition d on d.code=t.process_definition_code
-        join t_ds_project p on p.code=d.project_code
-        where 1 = 1
-        and t.is_sub_process = 0
+        where t.is_sub_process = 0
```

Comment:
@hiSandog @Radeity What about adding a foreign key in `t_ds_process_definition_code` referencing `project code` in `t_ds_project`?

Reply:
> @hiSandog @Radeity What about adding a foreign key in `t_ds_process_definition_code` referencing `project code` in `t_ds_project`?

This is a bit of a change.I am worried that some users may fail to upgrade

Reply:
> This is a bit of a change.I am worried that some users may fail to upgrade

Agree with you, and there's no foreign keys in other tables.

Reply:
> @hiSandog @Radeity What about adding a foreign key in `t_ds_process_definition_code` referencing `project code` in `t_ds_project`?

We don't need to add foreign key, the correct way is to add project code and project name in workflow instance, then we don't need to join the project table and workflow definition table in workflow instance listing page.

Reply:
> > @hiSandog @Radeity What about adding a foreign key in `t_ds_process_definition_code` referencing `project code` in `t_ds_project`?
> 
> We don't need to add foreign key, the correct way is to add project code and project name in workflow instance, then we don't need to join the project table and workflow definition table in workflow instance listing page.

This is a big change and requires extra work to ensure data consistency.
Submit another pr to change if necessary

Reply:
I will work on this.